kong/plugins/prometheus/exporter.lua

```diff
@@ -336,6 +368,38 @@ local function metric_data()
     enterprise.metric_data()
   end
 
+  -- Hybrid mode status
+  if cp_metrics then
+    -- Cleanup old metrics
+    metrics.data_plane_last_seen:reset()
+    metrics.data_plane_config_hash:reset()
+
+    for data_plane, err in kong.db.clustering_data_planes:each() do
+      if err then
+        kong.log.err("failed to list data planes: ", err)
+        goto next_data_plane
+      end
+
+      local labels = { data_plane.id, data_plane.hostname, data_plane.ip }
+
+      metrics.data_plane_last_seen:set(data_plane.last_seen, labels)
+      metrics.data_plane_config_hash:set(config_hash_to_number(data_plane.config_hash), labels)
+
+      labels[4] = data_plane.version
+      local compatible = 1
+
+      if data_plane.sync_status == CLUSTERING_SYNC_STATUS.KONG_VERSION_INCOMPATIBLE
+        or data_plane.sync_status == CLUSTERING_SYNC_STATUS.PLUGIN_SET_INCOMPATIBLE
+        or data_plane.sync_status == CLUSTERING_SYNC_STATUS.PLUGIN_VERSION_INCOMPATIBLE then
+
+        compatible = 0
+      end
```

Comment:
Question: What is the current (and plans for future) behavior of a Kong data plane when it figures out that it is incompatible with the control-plane it is connect to?
Does it continue to run or exits with an error?

Reply:
In this case DP will still connected to CP, but CP will not push config to DP. Generally the connection
will be closed only during handshake due to verification failure (on either side).